dtfs-central-api/src/services/repositories/utilisation-reports-repo/utilisation-reports-repo.ts

```diff
@@ -56,7 +56,7 @@ export const saveNotReceivedUtilisationReport = async (reportPeriod: ReportPerio
 
 export type GetUtilisationReportDetailsOptions = {
   reportPeriod?: UtilisationReport['reportPeriod'];
-  reportStatuses?: UtilisationReport['status'][];
+  excludeNotUploaded?: 'true' | 'false';
```

Comment:
Can this be a boolean? Was this because it's a query param? If so then it should be transformed from a string to a bool when passed to the repo as this detail should be abstracted away from the repo. If we do convert then it's probably worth creating a helper in common to do this conversion.

Reply:
I originally had this as a `boolean`, but changed it back because the `options` object when calling this function looked like
```typescript
const { excludeNotUploaded } = req.query; // undefined, true or false
...
{
  excludeNotUploaded: excludeNotUploaded && excludeNotUploaded === 'true',
}
```

and I didn't like it for some reason, but I am happy to switch it back to this

Reply:
I think it can just be
```
excludeNotUploaded: excludeNotUploaded === 'true',
```
?

Reply:
It could - my reasoning for having the first part is that if it is `undefined` (ie. if it was not in the query), then we pass in `undefined` to the repo method rather than `false` (I know this isn't true right now, but we may decide that passing in `false` does something different to `undefined` which I wanted to explicitly avoid)

Reply:
I'd argue YAGNI here. Also I'm not sure how `undefined` could be different from `false` when "excluding" something? If we want false to do something different we'd probably want a different name, and a different parameter type.

Reply:
Yeh, I was thinking YAGNI too. It's more the semantic side where the three possibilities are strictly speaking distinct things:
- 1: *do* exclude based on A
- 2: *do not* exclude based on A
- 3: *do nothing*

In this case I think it's fine to have options 2 and 3 mean the same thing, but my brain doesn't like conflating the two by default 😅